specutils/analysis/snr.py

```diff
@@ -0,0 +1,106 @@
+import numpy as np
+from ..utils import SpectralRegion
+
+__all__ = ['snr']
+
+
+def snr(spectrum, region=None, noise_region=None):
+    """
+    Calculate the mean S/N of the spectrum based on the flux and uncertainty in the spectrum. This will
+    be calculated over the regions, if they are specified.
+
+    Parameters
+    ----------
+    spectrum : ``specutils.spectra.spectrum1d.Spectrum1D``
+        The spectrum object overwhich the equivalent width will be calculated.
+
+    region: ``specutils.utils.SpectralRegion`` or list of ``specutils.utils.SpectralRegion``
+        Region within the spectrum to calculate the SNR.
+
+    noise_region: ``specutils.utils.SpectralRegion``
+        Region within the spectrum from which to calculate the noise (based on standard deviation
+        of the flux values)
+
+    Returns
+    -------
+    snr : float or list (based on region input)
+        Signal to noise ratio of the spectrum or within the regions
+
+    Notes
+    -----
+    The spectrum will need to have the uncertainty defined in order
+    for the SNR to be calculated.
+
+    Two methods implemented:
+      1) SNR based on uncertainty
+          This will compute the mean of the flux / uncertainty.
+
+      2) SNR based on noise_region
+          This will compute the standard devation of the flux in the noise_region
+          and the mean of the flux in the region and then return the mean flux divided
+          by the standard deviation of the noise_region flux.
+
+    """
+
+    # No region, therefore whole spectrum.
+    if region is None:
+        return _snr(spectrum, noise_region=noise_region)
+
+    # Single region
+    elif isinstance(region, SpectralRegion):
+        return _snr(spectrum, region=region, noise_region=noise_region)
+
+    # List of regions
+    elif isinstance(region, list):
+
+        snrs = []
+        for reg in region:
+            snrs.append(_snr(spectrum, region=reg, noise_region=noise_region))
```

Comment:
minor:  this would be *slightly* faster to do as a list comprehension.  I.e.:
``return [_snr(spectrum, region=reg, noise_region=noise_region) for reg in region]``

Reply:
modified. In next merge.